src/OTCNexus.sol

```diff
@@ -0,0 +1,283 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.13;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "./OtcMath.sol";
+
+error Router__InvalidTokenAmount();
+error Router__InvalidPriceAmount();
+error Router__InvalidPriceMultiplier();
+error Router__InvalidDeadline();
+error Router__InvalidTokenAddresses();
+error Router__InvalidTokenAndAmountMismatch();
+error Router__AllowanceToken0TooLow();
+error Router__AllowanceToken1TooLow();
+error Router__ApproveToken0Failed();
+error Router__DepositToken0Failed();
+error Router__RfsRemoved();
+error Router__Amount1TooHigh();
+error Router__TransferToken0Failed();
+error Router__TransferToken1Failed();
+error Router__NotMaker();
+error Router__RefundToken0Failed();
+error Router__OnlyFixedOrDynamicAllowed();
+error Router__OnlyFixedPriceOrAmountAllowed();
+error Router__OnlyOneTokenAccepted();
+
+/**
+ * @title OTCNexus
+ * @notice core swap functionality
+ */
+contract OTCNexus is Ownable {
+    uint256 public rfsIdCounter;
+    mapping(uint256 => RFS) private idToRfs;
+    mapping(address => RFS[]) private makerRfs;
+
+    struct RFS {
+        uint256 id;
+        address maker;
+        address token0;
+        address[] tokensAccepted;
+        uint256 amount0;
+        uint256 amount1;
+        uint256 usdPrice; // Implies the fixed price per token0
+        uint256 priceMultiplier;
+        uint256 deadline;
+        bool isDynamic;
+        bool isDeposited;
+        bool removed;
+    }
+
+    event RfsUpdated(uint256 rfsId, uint256 amount0, uint256 amount1, uint256 deadline);
+    event RfsFilled(uint256 rfsId, address taker, uint256 amount0, uint256 amount1);
+    event RfsRemoved(uint256 rfsId);
+    event RfsCreated(
+        uint256 rfsId,
+        address maker,
+        address indexed token0,
+        address[] indexed tokensAccepted,
+        uint256 amount0,
+        uint256 usdPrice,
+        uint8 priceMultiplier,
```

Comment:
priceMultiplier: uint8 [0-255] => value in % to apply to the market price.

priceMultiplier is a value that allows the user to set a specific dynamic % to follow the current price. i.e. 

1. -5% it will always have a discount of 5% on the current price. 
2. +5% it will always have a premium of 5% on the current price

uint8 it's because we allow maximum value to be between 0-255. In this way values between 0-100 means a discount, while values (100,255] implies a premium. i.e. 255 => 255% => 2.55x the current price.

Reply:
Ok, thanks for clarifying. It makes sense but let it be documented somewhere. If it approach is common in DeFi for priceMultiplier maybe a link with clarification would be better.
Also, if I understand correctly the value 0 would have the same result as 100 - ie. 100%, no premium nor discount, correct?

Reply:
No the value 0 means you give a discount of 100%. I am not sure if this is common in DeFi, this for me looks like an easy way to solve the problem